tpm2/tpm2_test.go

```diff
@@ -16,21 +16,41 @@ package tpm2
 
 import (
 	"bytes"
+	"context"
 	"crypto"
 	"crypto/ecdsa"
 	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/rsa"
 	"crypto/sha256"
 	"flag"
+	"io"
 	"math/big"
 	"os"
 	"reflect"
 	"testing"
 
 	"github.com/google/go-tpm/tpmutil"
+	"github.com/google/go-tpm/tpmutil/mssim"
 )
 
+var tpmMSSIM = flag.Bool("tpm_mssim", false, "Run integration tests against a Microsoft simulator.")
+
+func openTPM(t *testing.T) io.ReadWriteCloser {
+	if !*tpmMSSIM {
+		return openDeviceTPM(t)
+	}
+
+	conn, err := mssim.Open(context.Background(), mssim.Config{})
```

Comment:
I really wish this could be part of `tpmutil.OpenTPM`. But given it has two addresses i'm not sure how to do that.

Reply:
tpm2 does something like `mssim:host=localhost,port=2321`. That might be doable.

https://github.com/tpm2-software/tpm2-tools/blob/master/man/common/tcti.md#tcti-options